node/node.go

```diff
@@ -56,6 +57,7 @@ type Node struct {
 	// TODO full node) and in light, same thing `LightServices` (all expected services to be running in a light node.
 	// TODO `FullServices` can include `LightServices` + other services.
 	BlockServ *block.Service `optional:"true"`
+	ShareServ share.Service  // not optional
```

Comment:
I kinda don't like that one of the services is a pointer and then the other is an interface.

Reply:
Agree. I would generally prefer to have interfaces for each.